src/org/sosy_lab/common/log/BasicLogManager.java

```diff
@@ -414,7 +418,95 @@ private void log0(Level priority, StackTraceElement stackElement, String msg) {
     logger.log(record);
   }
 
-  private String buildMessageText(Object... args) {
+  /** {@inheritDoc} */
+  @Override
+  public void logUserException(Level priority, Throwable e, Object... args) {
+    checkNotNull(e);
+    checkNotNull(args);
+    @Var String additionalMessage = null;
+
+    if (wouldBeLogged(priority)) {
+      additionalMessage = buildAdditionalMessageText(args);
+      StringBuilder logMessage = buildUserExceptionLogMessage(priority, e, additionalMessage);
+      @Var StackTraceElement frame = locateStackTraceElement(e);
+
+      log0(priority, frame, logMessage.toString());
+    }
+    // Enforce lazy string building for DEBUG level
+    if (additionalMessage == null) {
+      logDebugException(e, args);
+    } else {
+      // but keep calculated results
+      logDebugException(e, additionalMessage);
+    }
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public void logfUserException(Level priority, Throwable e, String format, Object... args) {
+    checkNotNull(e);
+    checkNotNull(format);
+    checkNotNull(args);
+
+    @Var String additionalMessage = null;
+
+    if (wouldBeLogged(priority)) {
+      additionalMessage = formatString(format, args);
+      StringBuilder logMessage = buildUserExceptionLogMessage(priority, e, additionalMessage);
+      @Var StackTraceElement frame = locateStackTraceElement(e);
+
+      log0(priority, frame, logMessage.toString());
+    }
+    // Enforce lazy string building for DEBUG level
+    if (additionalMessage == null) {
+      logDebugException(e, args);
```

Comment:
Isn't `format` missing here? I am not sure this rather hard-to-understand logic is necessary. Maybe just have `wouldbeLogged(priority) || wouldbeLogged(EXCEPTION_DEBUG_LEVEL)` as condition for building the message?

Reply:
Yes I missed the format there.
That is a better solution. 
I will change that